community/lexicon/calendar/event.json

```diff
@@ -0,0 +1,137 @@
+                    },
+                    "endsAt": {
+                        "type": "string",
+                        "format": "datetime",
+                        "description": "Client-declared timestamp when this event ends."
+                    },
+                    "mode": {
+                        "type": "ref",
+                        "ref": "community.lexicon.calendar.event#mode",
+                        "description": "The attendance mode of the event."
+                    },
+                    "status": {
+                        "type": "ref",
+                        "ref": "community.lexicon.calendar.event#status",
+                        "description": "The status of the event."
+                    },
+                    "locations": {
+                        "type": "array",
+                        "description": "The locations where the event takes place.",
+                        "items": {
+                            "type": "union",
+                            "refs": [
+                                "community.lexicon.locations.address",
+                                "community.lexicon.locations.h3",
+                                "community.lexicon.locations.geo",
+                                "community.lexicon.locations.fsq",
+                                "community.lexicon.calendar.event#virtualLocation"
+                            ]
+                        }
+                    }
+                }
+            }
+        },
+        "mode": {
+            "type": "string",
+            "description": "The mode of the event.",
+            "default": "community.lexicon.calendar.event#inperson",
+            "knownValues": [
+                "community.lexicon.calendar.event#virtual",
+                "community.lexicon.calendar.event#inperson",
+                "community.lexicon.calendar.event#hybrid"
+            ]
+        },
+        "virtual": {
+            "type": "token",
+            "description": "A virtual event that takes place online."
+        },
+        "inperson": {
+            "type": "token",
+            "description": "An in-person event that takes place offline."
+        },
+        "hybrid": {
+            "type": "token",
+            "description": "A hybrid event that takes place both online and offline."
+        },
+        "status": {
+            "type": "string",
+            "description": "The status of the event.",
+            "default": "community.lexicon.calendar.event#scheduled",
+            "knownValues": [
+                "community.lexicon.calendar.event#scheduled",
+                "community.lexicon.calendar.event#rescheduled",
+                "community.lexicon.calendar.event#cancelled",
+                "community.lexicon.calendar.event#postponed"
+            ]
+        },
+        "scheduled": {
+            "type": "token",
+            "description": "The event has been created and scheduled."
```

Comment:
Curious on the differences or preferences around having tokens as fragments within a lexicon vs independent lexicon as shown in the ATProto lexicon traffic lights example (https://atproto.com/guides/lexicon#tokens)

Reply:
This is a pattern that I've taken from the bsky app lexicon that I've seen used pretty recently.

Example: https://github.com/bluesky-social/atproto/blob/main/lexicons/app/bsky/feed/defs.json#L228-L248